### PR TITLE
feat(smoke): deeper invariants + 500 battles per gen

### PR DESCRIPTION
**Pull request by uehlbran**
## Summary

- Increase `SMOKE_BATTLES_PER_GEN` from 200 to 500 for broader random battle coverage
- Add 5 new invariants in `checkBattleInvariants`: turn monotonicity, valid primary status, valid weather type, valid terrain type, damage/heal amount > 0
- Export `checkBattleInvariants` and `SmokeInvariantViolation` for unit testing
- Add 25 unit tests covering all invariant paths (empty stream, valid/invalid status, weather, terrain, turn order, damage amount, heal amount)

## Test plan

- [ ] `npx vitest run tools/oracle-validation/tests/smoke-runner.test.ts` — 25 tests pass
- [ ] `npm run verify:local` passes
- [ ] Smoke invariant checks: 2 → 7 (target: 7+)

Closes: N/A

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Increased smoke-test workload per generation.
  * Expanded invariant validation: stricter turn ordering, whitelisted status/weather/terrain values, and non-negative checks for damage/heal amounts; violation reports now reference event positions.

* **Tests**
  * Added comprehensive tests covering all new invariant checks with valid and invalid cases for turns, statuses, weather, terrain, damage, and healing.

### CODE REVIEW

File: tools/oracle-validation/src/smoke-runner.ts [hunk 1]
```diff
@@ -224,13 +228,36 @@ export function generateMinimalTeam(
...
   description: string;
 }
 
-function checkBattleInvariants(events: readonly BattleEvent[]): SmokeInvariantViolation[] {
+export function checkBattleInvariants(events: readonly BattleEvent[]): SmokeInvariantViolation[] {
   const violations: SmokeInvariantViolation[] = [];
+  let lastTurnNumber = 0;
 
   for (let i = 0; i < events.length; i++) {
     const event = events[i];
```
[the remark this answers is not in the document]
Comment (uehlbran) [reply tools/oracle-validation/src/smoke-runner.ts]: Fixed in 6fba1940 — replaced raw string literals with CORE_STATUS_IDS, CORE_WEATHER_IDS, and CORE_TERRAIN_IDS from @pokemon-lib-ts/core.

File: tools/oracle-validation/src/smoke-runner.ts [hunk 2]
```diff
@@ -224,13 +228,36 @@ export function generateMinimalTeam(
...
+  "heavy-rain",
+  "strong-winds",
+]);
+
+const VALID_TERRAIN_TYPES = new Set(["electric", "grassy", "psychic", "misty"]);
+
+export interface SmokeInvariantViolation {
+  eventIndex: number;
   description: string;
 }
```
[the remark this answers is not in the document]
Comment (uehlbran) [reply tools/oracle-validation/src/smoke-runner.ts]: Fixed in 6fba1940 — both fields of SmokeInvariantViolation are now readonly.
